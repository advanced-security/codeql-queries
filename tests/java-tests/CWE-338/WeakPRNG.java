import java.io.File;
import java.io.FileNotFoundException;
import java.util.Random;
import java.util.Scanner;

public class WeakPRNG {
    public static void main(String[] args) {
        File file = new File("input.txt");
        try {
            Scanner scanner = new Scanner(file);
            while (scanner.hasNextLine()) {
                String line = scanner.nextLine();
                int randomNumber = generateRandomNumber(line);
                int randomNumberWithEntropy = generateRandomNumberWithEntropy(line);
                System.out.println(line + " " + randomNumber + " " + randomNumberWithEntropy);
            }
            scanner.close();
        } catch (FileNotFoundException e) {
            System.out.println("File not found.");
        }
    }

    private static int generateRandomNumber(String line) {
        int seed = line.hashCode();
        Random random = new Random(seed);
        int randomNumber = random.nextInt();
        return randomNumber;
    }

    private static int generateRandomNumberWithEntropy(String line) {
        double randomDouble = Math.random();
        int randomNumberWithEntropy = (int) (randomDouble * 100);
        return randomNumberWithEntropy;
    }
}