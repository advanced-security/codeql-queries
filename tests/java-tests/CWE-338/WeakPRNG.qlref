CWE-338/WeakPRNG.ql