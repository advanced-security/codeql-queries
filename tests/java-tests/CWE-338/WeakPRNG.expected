| WeakPRNG.java:26:28:26:43 | nextInt(...) | Using weak PRNG |
| WeakPRNG.java:31:31:31:43 | random(...) | Using weak PRNG |
